AmplifyPlugins/Predictions/AWSPredictionsPlugin/Liveness/Service/FaceLivenessSession.swift

```diff
@@ -58,16 +62,31 @@ public final class FaceLivenessSession: LivenessService {
     ) {
         serverEventListeners[event] = listener
     }
+    
+    public func register(listener: @escaping (Challenge) -> Void, on event: LivenessEventKind.Server) {
+        challengeTypeListeners[event] = listener
+    }
 
     public func closeSocket(with code: URLSessionWebSocketTask.CloseCode) {
         websocket.close(with: code)
     }
 
-    public func initializeLivenessStream(withSessionID sessionID: String, userAgent: String = "") throws {
+    public func initializeLivenessStream(withSessionID sessionID: String, 
+                                         userAgent: String = "",
+                                         challenges: [Challenge]? = nil) throws {
         var components = URLComponents(url: baseURL, resolvingAgainstBaseURL: false)
+        let challengeList: [Challenge]
+        if let challenges = challenges {
+            challengeList = challenges
+        } else {
+            challengeList = FaceLivenessSession.supportedChallenges
+        }
         components?.queryItems = [
             URLQueryItem(name: "session-id", value: sessionID),
-            URLQueryItem(name: "challenge-versions", value: "FaceMovementAndLightChallenge_1.0.0"),
+            URLQueryItem(name: "precheck-view-enabled", value: options.preCheckViewEnabled ? "1":"0"),
+            URLQueryItem(name: "attempt-id", value: options.viewId),
```

Comment:
should `attempt-id` get changed to `view-id`?

Reply:
added a Todo to revisit this - pending confirmation